GVFS/GVFS.FunctionalTests/Tools/GitHelpers.cs

```diff
@@ -60,11 +60,21 @@ public static void CheckGitCommandAgainstGVFSRepo(string virtualRepoRoot, string
             string gvfsRepoRoot,
             string command,
             Dictionary<string, string> environmentVariables = null,
-            bool cleanErrors = true)
+            bool cleanErrors = true,
+            bool trimUpgradeReminder = true)
         {
             ProcessResult result = GitProcess.InvokeProcess(gvfsRepoRoot, command, environmentVariables);
 
             string errors = result.Errors;
+            if (trimUpgradeReminder)
+            {
+                string upgradeReminder = Environment.NewLine + "A new version of GVFS is available. Run `gvfs upgrade` to start the upgrade." + Environment.NewLine;
+                if (errors.EndsWith(upgradeReminder))
+                {
+                    errors = errors.Substring(0, errors.LastIndexOf(upgradeReminder));
+                }
+            }

```

Comment:
Could this be incorporated into the code below so you would have something like
```
IEnumerable<string> lines = errors.Split(new string[] { Environment.NewLine }, StringSplitOptions.None);
if (trimUpgradeReminder)
{
    lines = lines.Where(line => !line.StartsWith("A new version..."));
}

if (cleanErrors)
{
    lines = lines.Where(line => !line.StartsWith("Waiting for "));
}

errors = string.Join(Environment.NewLine, lines);
```

You could also check both flags and only do the `Split` and `Join` if at least one was set to avoid to perf hit when both are false.